src/firefly/js/visualize/ImageViewerLayout.jsx

```diff
@@ -316,19 +317,43 @@ function makeTileDrawers(pv) {
 
     const plot= primePlot(pv);
     const rootDrawer= (
-        <TileDrawerCanvas opacity={1} plot={plot} plotView={pv} key={'TileDrawer:'+pv.plotId} />
+        <ImageRender opacity={1} plot={plot} plotView={pv} key={'TileDrawer:'+pv.plotId} />
     );
     const drawers= pv.overlayPlotViews.filter( (opv) => opv.visible && opv.plot).map( (opv) => {
         return (
-            <TileDrawerCanvas opacity={opv.opacity} plot={opv.plot} plotView={pv}
-                key={'TileDrawer-overlay:'+opv.imageOverlayId}
+            <ImageRender opacity={opv.opacity} plot={opv.plot} plotView={pv}
+                         tileAttributes={opv.colorAttributes}
+                         shouldProcess={(im, newData, imState, nextImState) => {
+                                  if (newData) return imState.color!== imState.srcImageColor;
+                                  else         return imState.color!==nextImState.color;
+                              }}
+                         processor={makeMaskColorProcessor(opv.colorAttributes.color)}
+                         key={'TileDrawer-overlay:'+opv.imageOverlayId}
             />
         );
     });
     drawers.unshift(rootDrawer);
     return drawers;
 }
 
+
+function makeMaskColorProcessor(colorStr) {
+    return (imageData) => {
+        const cAry= Color.toRGB(colorStr);
+        const {data}= imageData;
+        const len= data.length;
+        for(let i= 0; i<len; i+=4) {
+            if (data[i] || data[i+1] || data[i+2]) {
```

Comment:
Does this mean if data[i], data[i+1] and data[i+2] are all zero, then the condition for 'if' is 'false'?